protected/models/User.php

```diff
@@ -618,7 +618,8 @@ public function saveFirms(array $firms)
         if (!$this->global_firm_rights && count($firms) === 0) {
             throw new FirmSaveException('When global firm rights are not set, a firm must be selected');
         }
-
+        //Set the has_selected_firms to be true
+        User::updateByPk($this->id, array("has_selected_firms"=>1));
```

Comment:
As Andrew pointed out, this should also be in the transaction that is started below, just in case there is an error when saving the FirmUserAssignments